src/main/java/net/minecraftforge/common/util/EnumHelper.java

```diff
@@ -141,6 +143,19 @@ public static EnumRarity addRarity(String name, TextFormatting color, String dis
     {
         return addEnum(EnumRarity.class, name, color, displayName);
     }
+    /**
+     * 
+     * @param name - The name the new enum constant should have
+     * @param textureLocation - The path to the texture for this armor type. Formatted as domain:path.
+     * @param hash - A short unique 3 letter name for the armor type 
```

Comment:
- If that hash must match on client and server or needs to be persistent across game loads:
   You would probably have to use a registry. I don't think enough mods will be adding horse armor to justify an advanced solution. Just have mods hard-code their values and negotiate collisions manually.
- If the hash is only used on the client:
   I think you could simply generate the hashes automatically starting with "aaa", "aab", etc. Just make sure they don't collide with vanilla's.